src/GraphClient.ts

```diff
@@ -0,0 +1,75 @@
+import ApolloClient from 'apollo-client';
+import { WebSocketLink } from 'apollo-link-ws';
+import { InMemoryCache } from 'apollo-cache-inmemory';
+import { SubscriptionClient } from 'subscriptions-transport-ws';
+import gql from 'graphql-tag';
+
+import WebSocket = require('ws');
+
+/**
+ * The class interacts with graph node server for subscription and query.
+ */
+export default class GraphClient {
+  private subgraphEndPoint: string;
+
+  /**
+   * GraphClient constructor.
+   *
+   * @param {string} subgraphEndPoint Subgraph http/ws end point.
+   */
+  public constructor(subgraphEndPoint: string) {
+    this.subgraphEndPoint = subgraphEndPoint;
+  }
+
+  /**
+   * Subscribes to the input subscription subgraph query and delegates the response
+   * to observer i.e. TransactionHandler.
+   * Documentation: https://www.apollographql.com/docs/react/advanced/subscriptions/
+   *
+   * @param subscriptionQry Subscription query object.
+   * @return `Subscription` Query subscription object
+   */
+  public subscribe(subscriptionQry: string) {
+    if( !subscriptionQry ) {
+      const err = new TypeError("Mandatory Parameter 'subscriptionQry' is missing or invalid.");
+      return Promise.reject(err);
```

Comment:
Why not throw error?
```suggestion
      throw err;
```

Reply:
AssertAsync works with Promise.reject else we will need to write custom callback to handle throw.  We have been using this in other repos also.

Reply:
Why is our implementation influenced by the unit test utility code?
It should be `throw` instead of `promise.reject`. This is not async function.